### ISSUE TITLE
Wrong TTS url

### ISSUE THREAD

**Issue by KeltenP** [AlexxIT/WebRTC#742]
go2rtc version: 1.9.4
webrtc integration: 3.5.2
Home assistant: 2024.8.3
Reolink firmware: v3.0.0.3308_2407315182

I am trying to send TTS to my Reolink WiFi Doorbell. I already have 2 way audio working fine.

I am running go2rtc 1.9.4 inside my frigate instance. In home assistant, I have the webrtc integration added and it points to my local go2rtc url (http://172.16.1.66:1984).

In go2rtc, my doorbell stream for sending audio is:

```
doorbell_send_audio:
      - rtsp://{FRIGATE_REOLINK_USERNAME}:{FRIGATE_REOLINK_PASSWORD}@{FRIGATE_DOORBELL_IP}:554/Preview_01_sub
      - ffmpeg:doorbell_send_audio#audio=opus
```

In my home assistant configuration yaml, I have:

```
media_player:
  - platform: webrtc
    name: doorbell_tts
    stream: doorbell_send_audio
    audio: pcma
```

Next, I use tts.speak to send tts audio to the camera. I am using the cache option so really it is sending a mp3 file from my home assistant server.

When I send TTS, I get no audio at all and see this in the go2rtc logs:

```
[api] POST /api/streams?dst=doorbell_send_audio&src=ffmpeg:http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3%23audio%3Dpcma%23input%3Dfile 172.16.1.123:42466
```

Something is encoding the ffmpeg options at the end of the url (# -> %23, = -> %3D) which seems to be the problem.

If I manually try to POST to this url using curl:

```
curl --verbose --request POST --url 'http://172.16.1.66:1984/api/ffmpeg?dst=doorbell_send_audio&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3%23audio%3Dpcma%23input%3Dfile'
```
I get the error ` ffmpeg: unsupported params: #audio=pcma#input=file#audio=auto#input=file `

If I change the **%23** after `mp3` to **%2523**, it works and I can hear the TTS. (the audio is choppy and cuts out, but atleast I can hear it)

```
curl --verbose --request POST --url 'http://172.16.1.66:1984/api/ffmpeg?dst=doorbell_send_audio&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3%2523audio%3Dpcma%23input%3Dfile'
```

Or, if I remove the encoding entirely and just put the normal characters back, it also works.
```
curl --verbose --request POST --url 'http://172.16.1.66:1984/api/ffmpeg?dst=doorbell_send_audio&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3#audio=pcma#input=file'
```

What is causing the ffmpeg parameters to be encoded? go2rtc or webrtc?

Why does changing %23 to %2523 before the ffmpeg parameters make it work?

Not sure if this is a webrtc issue or go2rtc issue (or something else?)

**Comment by AlexxIT** [comment]
1. Your IP different - 172.16.1.123 and 172.16.1.66
2. Your API different - `/api/streams` and `/api/ffmpeg`. FFmpeg API takes file link as input. Streams API takes stream source as input.

**Comment by KeltenP** [comment]
172.16.1.66:1984 is my go2rtc instance. All /api/ requests go to this address. 
172.16.1.123:8123 is my home assistant. This is where the mp3 file is.

Sorry, I never realized I was mixing up /api/streams and /api/ffmpeg. My mistake.

When curling the /api/streams, I was getting "no consumer found". I think because the codec didn't match. From go2rtc, the codec that shows up as a consumer is "audio, sendonly, PCMU/8000"

In home assistant configuration.yaml, I changed **pcma** to **pcmu**.
```
media_player:
  - platform: webrtc
    name: doorbell_tts
    stream: doorbell_send_audio
    audio: pcmu
```

Now I am getting audio out using the doorbell_tts.

And it works using manual curl:
```
curl --verbose --request POST --url "http://172.16.1.66:1984/api/streams?dst=doorbell_send_audio&src=ffmpeg:http://172.16.1.123:8123/api/tts_proxy/94aab1c7563c12d0fbba2d2f4d76e16b1caada9e_en-ca_f579540522_tts.home_assistant_cloud.mp3%23audio%3Dpcmu%23input%3Dfile"
```

It is still choppy/cuts out but that is a different issue.

**Comment by AlexxIT** [comment]
What about `/api/ffmpes`? Is it works better with right codec?

**Comment by KeltenP** [comment]
I tried the follwing /api/ffmpeg commands:

No extra parameters:
```
curl --verbose --request POST --url "http://172.16.1.66:1984/api/ffmpeg?dst=doorbell_send_audio&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3"
```

With pcmu parameter:
```
curl --verbose --request POST --url "http://172.16.1.66:1984/api/ffmpeg?dst=doorbell_send_audio&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3#audio=pcmu"
```

With pcmu and file parameter:
```
curl --verbose --request POST --url "http://172.16.1.66:1984/api/ffmpeg?dst=doorbell_send_audio&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3#audio=pcmu#input=file"
```

All of these options work.



However, encoding the parameters does NOT work:

```
curl --verbose --request POST --url "http://172.16.1.66:1984/api/ffmpeg?dst=doorbell_send_audio&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3%23audio%3Dpcmu%23input%3Dfile"

ffmpeg: unsupported params: #audio=pcmu#input=file#audio=auto#input=file
```

**Comment by AlexxIT** [comment]
`/api/ffmpeg` uses autoselect for codec. And `file` flag you already set via `file` query param.
What about quality?
> the audio is choppy and cuts out

**Comment by KeltenP** [comment]
I did some more expirementing. The /api/ffmpeg is working very well with the following configuration:

```
doorbell:
  - ffmpeg:http://{FRIGATE_DOORBELL_IP}/flv?port=1935&app=bcs&stream=channel0_main.bcs&user={FRIGATE_REOLINK_USERNAME}&password={FRIGATE_REOLINK_PASSWORD}#video=copy#audio=copy#audio=opus
  - rtsp://{FRIGATE_REOLINK_USERNAME}:{FRIGATE_REOLINK_PASSWORD}@{FRIGATE_DOORBELL_IP}:554/Preview_01_sub      
```

doorbell is the main stream and RTSP backchannel. I removed `doorbell_send_audio` because it didn't seem to provide any benefit.

With this setup, I run the following:
```
curl --verbose --request POST --url "http://172.16.1.66:1984/api/ffmpeg?dst=doorbell&file=http://172.16.1.123:8123/api/tts_proxy/b06d94ea6efe2906bfc7273c49f6ea1f4652fc40_en-us_6e8b81ac47_tts.home_assistant_cloud.mp3"
```
And the audio sounds perfect, there is no choppiness or cut outs.

HOWEVER, I need to make sure that I do not have another process using the backchannel. If I open my frigate card in home assistant and connect the microphone, then try to use the /api/ffmpeg again, the audio becomes choppy and cuts out again.

After I close the frigate card and disconnect microphone access, then /api/ffmpeg gives clear audio again. I guess the audio streams are interfering. I assume reolink only has single backchannel.

Is it possible to resolve this interference some way? Pipe the /api/ffmpeg to an exisiting audio channel? I don't know how difficult this would be. Or maybe there is a better way to set up my configuration.

Thanks for your help with this.
